### PR TITLE
Add user authentication and set up docker containers

### PR DESCRIPTION
**Pull request by bilalebi**

### CODE REVIEW

File: LICENSE
Comment (nerdstrike) [inline LICENSE:182]: I believe it says "Don't include the brackets"...

File: NOTICE
Comment (nerdstrike) [inline NOTICE]: 2018 is an interesting starting year

File: requirements.txt
Comment (nerdstrike) [inline requirements.txt:1]: I'm not a big fan of fixing library versions until it becomes necessary to do so (mainly to stop long term incompatibilities cropping up when another dependency needs to be bumped). Your call.

File: requirements.txt
Comment (nerdstrike) [inline requirements.txt:16]: Surely you don't need six for a pure Python 3 project?

File: thr/user/tests.py
Comment (nerdstrike) [inline thr/user/tests.py:97]: Don't understand what this is testing. Please comment

**Review by nerdstrike — CHANGES_REQUESTED** [review]
A lack of Django familiarity means I can't usefully review the plumbing, but I've spotted some obvious things.

File: requirements.txt
Comment (bilalebi) [reply requirements.txt:1]: I used `pip freeze > requirements.txt` which create a snapshot in time of the packages used their versions 
I think both approaches have their pros and cons, adding library's version guarantee that we won't end up in dependency hell.. some versions of packages will not work well with other packages and their respective versions. It is hard to predict how changes in the future will affect these interrelationships.

File: requirements.txt
Comment (bilalebi) [reply requirements.txt:16]: For some reason, it's required by `packaging` which is required by `pytest`
```
$ pip show packaging    
Name: packaging
Version: 20.4
Summary: Core utilities for Python packages
Home-page: https://github.com/pypa/packaging
Author: Donald Stufft and individual contributors
Author-email: donald@stufft.io
License: BSD-2-Clause or Apache-2.0
Location: /Users/username/git/thr/thr_env/lib/python3.8/site-packages
Requires: six, pyparsing
Required-by: pytest
```

Comment (nerdstrike) [reply requirements.txt:16]: I suppose I'm surprised that pip doesn't take care of that.

Comment (bilalebi) [reply requirements.txt:16]: The `requirements.txt` itself is generated using pip 
Just to make sure `six` is needed I uninstalled it and run `pytest`, then I got the error:
```
File "/Users/username/git/thr/thr_env/lib/python3.8/site-packages/pytest_django/live_server_helper.py", line 1, in <module>
    import six
ModuleNotFoundError: No module named 'six'
```

File: thr/user/tests.py
Comment (bilalebi) [reply thr/user/tests.py:97]: Done, let me if it's clear

File: LICENSE
Comment (bilalebi) [reply LICENSE:182]: Yes.. it says _Don't include the brackets_, I don't see any bracket in the source code

File: requirements.txt
Comment (nerdstrike) [reply requirements.txt:1]: Ah, now I remember why I hate it. You can't use a shared python env without it messing with other projects.

**Review by nerdstrike — APPROVED** [review]
All my superficial concerns are addressed. I recommend you wait for @sanjay-boddu 's approval before merging.

**Pull request merged**
